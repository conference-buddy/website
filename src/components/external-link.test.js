import React from 'react'
import { shallow } from 'enzyme'
import ExternalLink from './external-link'

const testUrl = 'http://conferencebuddy.io/'
const classTextLink = 'TextLink'
const linkText = 'Conference Buddy'
const linkImage = <img src="image.jpg" />

const additionalClassLinkPrimary = 'TextLink-primary'
const additionalClassLinkSecondary = 'TextLink-secondary'

const requiredProps = {
  url: testUrl,
  primary: true,
  secondary: false,
}

describe('<ExternalLink />', () => {
  it('renders without crashing', () => {
    shallow(<ExternalLink {...requiredProps}>{linkText}</ExternalLink>)
  })

  describe('renders a link with an url and a fixed class', () => {
    const wrapper = shallow(
      <ExternalLink url={testUrl}>{linkText}</ExternalLink>
    )
    it('renders a link', () => {
      expect(wrapper.find('a')).toHaveLength(1)
    })
    it('renders a link with a url', () => {
      expect(wrapper.find('a').prop('href')).toEqual(testUrl)
    })
    it('renders a link with a class', () => {
      expect(wrapper.hasClass(classTextLink)).toBe(true)
    })
  })

  describe('renders an a-element with a given content', () => {
    it('renders a link with a text as children', () => {
      const wrapper = shallow(
        <ExternalLink url={testUrl}>{linkText}</ExternalLink>
      )
      expect(wrapper.text()).toBe(linkText)
      expect(wrapper).toMatchSnapshot()
    })
    it('renders a link with an image as children', () => {
      const wrapper = shallow(
        <ExternalLink url={testUrl}>{linkImage}</ExternalLink>
      )
      expect(wrapper.contains(linkImage)).toBe(true)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('adds a additional class dependend on props', () => {
    it('adds a class for style primary', () => {
      const wrapper = shallow(
        <ExternalLink url={testUrl} primary>
          {' '}
          {linkText}
        </ExternalLink>
      )
      expect(wrapper.hasClass(additionalClassLinkPrimary)).toBe(true)
      expect(wrapper.hasClass(additionalClassLinkSecondary)).toBe(false)
      expect(wrapper).toMatchSnapshot()
    })
    it('adds a class for style secondary', () => {
      const wrapper = shallow(
        <ExternalLink url={testUrl} secondary>
          {linkText}
        </ExternalLink>
      )
      expect(wrapper.hasClass(additionalClassLinkSecondary)).toBe(true)
      expect(wrapper.hasClass(additionalClassLinkPrimary)).toBe(false)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
