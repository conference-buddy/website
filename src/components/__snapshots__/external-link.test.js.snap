// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExternalLink /> adds a additional class dependend on props adds a class for style primary 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExternalLink
    primary={true}
    url="http://conferencebuddy.io/"
  >
     
    Conference Buddy
  </ExternalLink>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        " ",
        "Conference Buddy",
      ],
      "className": "TextLink TextLink-primary ",
      "href": "http://conferencebuddy.io/",
    },
    "ref": null,
    "rendered": Array [
      " ",
      "Conference Buddy",
    ],
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          " ",
          "Conference Buddy",
        ],
        "className": "TextLink TextLink-primary ",
        "href": "http://conferencebuddy.io/",
      },
      "ref": null,
      "rendered": Array [
        " ",
        "Conference Buddy",
      ],
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<ExternalLink /> adds a additional class dependend on props adds a class for style secondary 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExternalLink
    secondary={true}
    url="http://conferencebuddy.io/"
  >
    Conference Buddy
  </ExternalLink>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Conference Buddy",
      "className": "TextLink  TextLink-secondary",
      "href": "http://conferencebuddy.io/",
    },
    "ref": null,
    "rendered": "Conference Buddy",
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Conference Buddy",
        "className": "TextLink  TextLink-secondary",
        "href": "http://conferencebuddy.io/",
      },
      "ref": null,
      "rendered": "Conference Buddy",
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<ExternalLink /> renders an a-element with a given content renders a link with a text as children 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExternalLink
    url="http://conferencebuddy.io/"
  >
    Conference Buddy
  </ExternalLink>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Conference Buddy",
      "className": "TextLink  ",
      "href": "http://conferencebuddy.io/",
    },
    "ref": null,
    "rendered": "Conference Buddy",
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Conference Buddy",
        "className": "TextLink  ",
        "href": "http://conferencebuddy.io/",
      },
      "ref": null,
      "rendered": "Conference Buddy",
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<ExternalLink /> renders an a-element with a given content renders a link with an image as children 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExternalLink
    url="http://conferencebuddy.io/"
  >
    <img
      src="image.jpg"
    />
  </ExternalLink>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        src="image.jpg"
      />,
      "className": "TextLink  ",
      "href": "http://conferencebuddy.io/",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "src": "image.jpg",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          src="image.jpg"
        />,
        "className": "TextLink  ",
        "href": "http://conferencebuddy.io/",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "image.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
